packages/@react-aria/dnd/src/constants.ts

```diff
@@ -31,28 +46,52 @@ export const DROP_OPERATION_ALLOWED = {
   uninitialized: DROP_OPERATION.all
 };
 
-export const EFFECT_ALLOWED = invert(DROP_OPERATION_ALLOWED);
+interface EffectAllowed {
+  0: 'none' | 'cancel',
+  1: 'move',
+  2: 'copy',
+  3: 'copyMove',
+  4: 'link',
+  5: 'linkMove',
+  6: 'copyLink',
+  7: 'all'
+}
+
+export const EFFECT_ALLOWED: EffectAllowed = invert(DROP_OPERATION_ALLOWED) as unknown as EffectAllowed;
 EFFECT_ALLOWED[DROP_OPERATION.all] = 'all'; // ensure we don't map to 'uninitialized'.
 
-export const DROP_EFFECT = invert(DROP_OPERATION);
+interface DropEffect {
+  0: 'none' | 'cancel',
+  1: 'move',
+  2: 'copy',
+  4: 'link',
+  7: 'all'
+}
+export const DROP_EFFECT: DropEffect = invert(DROP_OPERATION) as unknown as DropEffect;
 export const DROP_EFFECT_TO_DROP_OPERATION: {[name: string]: DropOperation} = {
   none: 'cancel',
   link: 'link',
   copy: 'copy',
   move: 'move'
 };
 
-export const DROP_OPERATION_TO_DROP_EFFECT = invert(DROP_EFFECT_TO_DROP_OPERATION);
+interface DropOperationToDropEffect {
+  'cancel': 'none',
+  'link': 'link',
+  'copy': 'copy',
+  'move': 'move'
+}
+export const DROP_OPERATION_TO_DROP_EFFECT: DropOperationToDropEffect = invert(DROP_EFFECT_TO_DROP_OPERATION) as unknown as DropOperationToDropEffect;
```

Comment:
bit unfortunate that you have to explicitly set up this interface manually inverting it